File: salsel-backend/src/main/java/com/salsel/dto/BillingDto.java
package com.salsel.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class BillingDto {
    private Long id;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate invoiceDate;
    private Long customerAccountNumber;
    private Long airwayBillNo;
    private String invoiceNo;
    private String address;
    private String country;
    private String city;
    private String taxNo;
    private String taxInvoiceTo;
    private String invoiceType;
    private String customerRef;
    private String product;
    private String serviceDetails;
    private Double charges;
    private Boolean status;
    private String billingStatus;
    private Double vatTax;
    private Double taxAmount;
    private Boolean isEmailSend;
}

File: salsel-ui/src/app/components/billing/model/billingDto.ts
export interface IBilling {
  invoiceDate?: any;
  customerAccountNumber?: string;
  airwayBillNo?: number;
  invoiceNo?: number;
  address?: string;
  country?: string;
  city?: string;
  taxNo?: number;
  taxInvoiceTo?: string;
  invoiceType?: string;
  customerRef?: number;
  product?: string;
  serviceDetails?: string;
  charges?: number;
  status?: boolean;
  billingStatus?: string;
  vatTax?: number;
  customerName?:string;
}

export class Billing implements IBilling {
  invoiceDate?: any;
  customerAccountNumber?: string;
  airwayBillNo?: number;
  invoiceNo?: number;
  address?: string;
  country?: string;
  city?: string;
  taxNo?: number;
  taxInvoiceTo?: string;
  invoiceType?: string;
  customerRef?: number;
  product?: string;
  serviceDetails?: string;
  charges?: number;
  status?: boolean;
  billingStatus?: string;
  vatTax?: number;
  customerName?: string;
}
